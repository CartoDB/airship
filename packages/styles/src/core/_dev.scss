@import 'tools/_mixins';
@import 'variables/_variables';
