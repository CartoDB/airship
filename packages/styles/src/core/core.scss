@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
