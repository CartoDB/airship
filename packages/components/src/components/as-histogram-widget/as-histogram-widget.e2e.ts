import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-histogram-widget', () => {
  let page: E2EPage;

  beforeEach(async () => {
    page = await newE2EPage({ html: '<as-histogram-widget></as-histogram-widget>' });
  });

  describe('Rendering', () => {
    it('should render properly', async () => {
      const element: E2EElement = await page.find('as-histogram-widget');
      element.setProperty('heading', 'Histogram Widget Example');
      element.setProperty('description', 'Description for Histogram Widget');
      element.setProperty('showClear', true);
      element.setProperty('data', histogramData);
      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should not render header when showHeader is false', async () => {
      const element: E2EElement = await page.find('as-histogram-widget');
      element.setProperty('heading', 'Histogram Widget Example');
      element.setProperty('description', 'Description for Histogram Widget');
      element.setProperty('showHeader', false);
      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should render clear button', async () => {
      const element: E2EElement = await page.find('as-histogram-widget');
      element.setProperty('showClear', true);
      element.setProperty('data', histogramData);
      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should render selection properly', async () => {
      const element: E2EElement = await page.find('as-histogram-widget');
      element.setProperty('data', histogramData);
      await page.waitForChanges();

      element.callMethod('setSelection', [[0, 20]]);
      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should render colors properly', async () => {
      const element: E2EElement = await page.find('as-histogram-widget');
      element.setProperty('color', '#FFAAAA');
      element.setProperty('selectedColor', '#EEFFFF');
      element.setProperty('data', histogramData);
      await page.waitForChanges();

      await element.callMethod('setSelection', [[0, 20]]);

      expect(element.outerHTML).toMatchSnapshot();
    });
  });

  describe('Behaviour', () => {
    it('should emit an event with the selected range', async () => {
      const selectionChangedSpy = await page.spyOnEvent('selectionChanged');

      const element: E2EElement = await page.find('as-histogram-widget');
      element.setProperty('data', histogramData);
      await page.waitForChanges();

      await element.callMethod('setSelection', [0, 20]);
      await page.waitForChanges();

      expect(selectionChangedSpy).toHaveReceivedEventDetail([0, 20]);
    });

    it('should clear the selection', async () => {
      const element: E2EElement = await page.find('as-histogram-widget');
      element.setProperty('data', histogramData);
      await page.waitForChanges();

      element.callMethod('setSelection', [0, 20]);

      const currentSelection = await element.callMethod('getSelection');
      expect(currentSelection).toEqual([0, 20]);

      element.callMethod('clearSelection');

      const newSelection = await element.callMethod('getSelection');
      expect(newSelection).toEqual(null);
    });

    it('should adjust the selection to the closest buckets', async () => {
      const element: E2EElement = await page.find('as-histogram-widget');
      element.setProperty('data', histogramData);
      await page.waitForChanges();

      element.callMethod('setSelection', [4, 22]);

      const currentSelection = await element.callMethod('getSelection');
      expect(currentSelection).toEqual([0, 20]);
    });
  });
});

const histogramData = [
  { start: 0, end: 10, value: 5 },
  { start: 10, end: 20, value: 10 },
  { start: 20, end: 30, value: 15 },
  { start: 30, end: 40, value: 20 },
];
