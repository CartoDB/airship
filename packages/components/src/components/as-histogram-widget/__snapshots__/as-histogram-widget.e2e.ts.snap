// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-histogram-widget Rendering should not render header when showHeader is false 1`] = `"<as-histogram-widget class=\\"hydrated\\"><div class=\\"as-histogram-widget__wrapper\\"><span role=\\"tooltip\\" class=\\"as-histogram-widget__tooltip\\"></span><svg><g class=\\"yAxis\\" transform=\\"translate(30, 5)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"end\\"></g><g class=\\"xAxis\\" transform=\\"translate(30, 130)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\"><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">0</text></g></g><g transform=\\"translate(30, 5)\\"></g><g class=\\"brush\\" fill=\\"none\\" pointer-events=\\"all\\"><rect class=\\"overlay\\" pointer-events=\\"all\\" cursor=\\"crosshair\\" x=\\"30\\" y=\\"5\\" width=\\"764\\" height=\\"125\\"></rect><rect class=\\"selection\\" cursor=\\"move\\" fill=\\"#777\\" fill-opacity=\\"0.3\\" stroke=\\"#fff\\" shape-rendering=\\"crispEdges\\"></rect><rect class=\\"handle handle--e\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle handle--w\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><line class=\\"bottomline\\" stroke=\\"var(--as-color-complementary, #47DB99)\\" stroke-width=\\"4\\" y1=\\"130\\" y2=\\"130\\" pointer-events=\\"none\\"></line></g></svg></div></as-histogram-widget>"`;

exports[`as-histogram-widget Rendering should render clear button 1`] = `"<as-histogram-widget class=\\"hydrated\\"><h2 class=\\"as-histogram-widget__header\\"></h2><p class=\\"as-histogram-widget__description as-body\\"></p><div class=\\"as-histogram-widget__wrapper\\"><span role=\\"tooltip\\" class=\\"as-histogram-widget__tooltip\\"></span><svg><g class=\\"yAxis\\" transform=\\"translate(30, 5)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"end\\"><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,125.5)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,94.25)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">5</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,63)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,31.75)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">15</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,0.5)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">20</text></g></g><g class=\\"xAxis\\" transform=\\"translate(30, 130)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\"><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(186.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(372.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">20</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(558.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">30</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(744.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">40</text></g></g><g transform=\\"translate(30, 5)\\"><rect class=\\"bar\\" y=\\"125\\" x=\\"0\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"186\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"372\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"558\\" width=\\"185\\" height=\\"0\\"></rect></g><g class=\\"brush\\" fill=\\"none\\" pointer-events=\\"all\\"><rect class=\\"overlay\\" pointer-events=\\"all\\" cursor=\\"crosshair\\" x=\\"30\\" y=\\"5\\" width=\\"764\\" height=\\"125\\"></rect><rect class=\\"selection\\" cursor=\\"move\\" fill=\\"#777\\" fill-opacity=\\"0.3\\" stroke=\\"#fff\\" shape-rendering=\\"crispEdges\\"></rect><rect class=\\"handle handle--e\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle handle--w\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><line class=\\"bottomline\\" stroke=\\"var(--as-color-complementary, #47DB99)\\" stroke-width=\\"4\\" y1=\\"130\\" y2=\\"130\\" pointer-events=\\"none\\"></line></g></svg></div><button class=\\"as-btn as-btn--primary as-btn--s as-histogram-widget__clear\\">Clear selection</button></as-histogram-widget>"`;

exports[`as-histogram-widget Rendering should render colors properly 1`] = `"<as-histogram-widget class=\\"hydrated\\"><h2 class=\\"as-histogram-widget__header\\"></h2><p class=\\"as-histogram-widget__description as-body\\"></p><div class=\\"as-histogram-widget__wrapper\\"><span role=\\"tooltip\\" class=\\"as-histogram-widget__tooltip\\"></span><svg><g class=\\"yAxis\\" transform=\\"translate(30, 5)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"end\\"><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,125.5)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,94.25)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">5</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,63)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,31.75)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">15</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,0.5)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">20</text></g></g><g class=\\"xAxis\\" transform=\\"translate(30, 130)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\"><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(186.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(372.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">20</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(558.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">30</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(744.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">40</text></g></g><g transform=\\"translate(30, 5)\\"><rect class=\\"bar\\" y=\\"125\\" x=\\"0\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"186\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"372\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"558\\" width=\\"185\\" height=\\"0\\"></rect></g><g class=\\"brush\\" fill=\\"none\\" pointer-events=\\"all\\"><rect class=\\"overlay\\" pointer-events=\\"all\\" cursor=\\"crosshair\\" x=\\"30\\" y=\\"5\\" width=\\"764\\" height=\\"125\\"></rect><rect class=\\"selection\\" cursor=\\"move\\" fill=\\"#777\\" fill-opacity=\\"0.3\\" stroke=\\"#fff\\" shape-rendering=\\"crispEdges\\"></rect><rect class=\\"handle handle--e\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle handle--w\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><line class=\\"bottomline\\" stroke=\\"var(--as-color-complementary, #47DB99)\\" stroke-width=\\"4\\" y1=\\"130\\" y2=\\"130\\" pointer-events=\\"none\\"></line></g></svg></div></as-histogram-widget>"`;

exports[`as-histogram-widget Rendering should render properly 1`] = `"<as-histogram-widget class=\\"hydrated\\"><h2 class=\\"as-histogram-widget__header\\">Histogram Widget Example</h2><p class=\\"as-histogram-widget__description as-body\\">Description for Histogram Widget</p><div class=\\"as-histogram-widget__wrapper\\"><span role=\\"tooltip\\" class=\\"as-histogram-widget__tooltip\\"></span><svg><g class=\\"yAxis\\" transform=\\"translate(30, 5)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"end\\"><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,125.5)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,94.25)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">5</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,63)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,31.75)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">15</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,0.5)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">20</text></g></g><g class=\\"xAxis\\" transform=\\"translate(30, 130)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\"><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(186.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(372.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">20</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(558.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">30</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(744.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">40</text></g></g><g transform=\\"translate(30, 5)\\"><rect class=\\"bar\\" y=\\"125\\" x=\\"0\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"186\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"372\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"558\\" width=\\"185\\" height=\\"0\\"></rect></g><g class=\\"brush\\" fill=\\"none\\" pointer-events=\\"all\\"><rect class=\\"overlay\\" pointer-events=\\"all\\" cursor=\\"crosshair\\" x=\\"30\\" y=\\"5\\" width=\\"764\\" height=\\"125\\"></rect><rect class=\\"selection\\" cursor=\\"move\\" fill=\\"#777\\" fill-opacity=\\"0.3\\" stroke=\\"#fff\\" shape-rendering=\\"crispEdges\\"></rect><rect class=\\"handle handle--e\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle handle--w\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><line class=\\"bottomline\\" stroke=\\"var(--as-color-complementary, #47DB99)\\" stroke-width=\\"4\\" y1=\\"130\\" y2=\\"130\\" pointer-events=\\"none\\"></line></g></svg></div><button class=\\"as-btn as-btn--primary as-btn--s as-histogram-widget__clear\\">Clear selection</button></as-histogram-widget>"`;

exports[`as-histogram-widget Rendering should render selection properly 1`] = `"<as-histogram-widget class=\\"hydrated\\"><h2 class=\\"as-histogram-widget__header\\"></h2><p class=\\"as-histogram-widget__description as-body\\"></p><div class=\\"as-histogram-widget__wrapper\\"><span role=\\"tooltip\\" class=\\"as-histogram-widget__tooltip\\"></span><svg><g class=\\"yAxis\\" transform=\\"translate(30, 5)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"end\\"><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,125.5)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,94.25)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">5</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,63)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,31.75)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">15</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,0.5)\\"><line stroke=\\"currentColor\\" x2=\\"744\\"></line><text fill=\\"currentColor\\" x=\\"-10\\" dy=\\"0.32em\\">20</text></g></g><g class=\\"xAxis\\" transform=\\"translate(30, 130)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\"><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(186.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(372.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">20</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(558.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">30</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(744.5,0)\\"><text fill=\\"currentColor\\" y=\\"10\\" dy=\\"0.71em\\">40</text></g></g><g transform=\\"translate(30, 5)\\"><rect class=\\"bar\\" y=\\"125\\" x=\\"0\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"186\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"372\\" width=\\"185\\" height=\\"0\\"></rect><rect class=\\"bar\\" y=\\"125\\" x=\\"558\\" width=\\"185\\" height=\\"0\\"></rect></g><g class=\\"brush\\" fill=\\"none\\" pointer-events=\\"all\\"><rect class=\\"overlay\\" pointer-events=\\"all\\" cursor=\\"crosshair\\" x=\\"30\\" y=\\"5\\" width=\\"764\\" height=\\"125\\"></rect><rect class=\\"selection\\" cursor=\\"move\\" fill=\\"#777\\" fill-opacity=\\"0.3\\" stroke=\\"#fff\\" shape-rendering=\\"crispEdges\\"></rect><rect class=\\"handle handle--e\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle handle--w\\" cursor=\\"ew-resize\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><rect class=\\"handle--custom\\" fill=\\"var(--as-color-complementary, #47DB99)\\" cursor=\\"ew-resize\\" width=\\"15\\" height=\\"15\\" rx=\\"100\\" ry=\\"100\\"></rect><line class=\\"bottomline\\" stroke=\\"var(--as-color-complementary, #47DB99)\\" stroke-width=\\"4\\" y1=\\"130\\" y2=\\"130\\" pointer-events=\\"none\\"></line></g></svg></div></as-histogram-widget>"`;
